motionplan/motionPlanner_test.go

```diff
@@ -469,6 +476,72 @@ func TestSolverFrameGeometries(t *testing.T) {
 	}
 }
 
+func TestArmConstraintSpecificationSolve(t *testing.T) {
+	fs := makeTestFS(t)
+	positions := frame.StartPositions(fs)
+	worldState := &frame.WorldState{}
+	constraints := &motionpb.Constraints{}
+
+	checkReachable := func(worldState *frame.WorldState, constraints *motionpb.Constraints) error {
+		goal1 := spatialmath.NewPose(r3.Vector{X: 600, Y: 100, Z: 300}, &spatialmath.OrientationVectorDegrees{OX: 1})
+		_, err := PlanMotion(
+			context.Background(),
+			logger.Sugar(),
+			frame.NewPoseInFrame("gantryY", goal1),
+			fs.Frame("xArmVgripper"),
+			positions,
+			fs,
+			worldState,
+			constraints,
+			nil,
+		)
+		return err
+	}
+
+	// Verify that the goal position is reachable with no obstacles
+	err := checkReachable(worldState, constraints)
+	test.That(t, err, test.ShouldBeNil)
+
+	// Add an obstacle
+	box, err := spatialmath.NewBox(spatialmath.NewPoseFromPoint(r3.Vector{350, 0, 0}), r3.Vector{0, 8000, 8000}, "theWall")
+	test.That(t, err, test.ShouldBeNil)
+	worldState = &frame.WorldState{
+		Obstacles: []*frame.GeometriesInFrame{frame.NewGeometriesInFrame("gantryY", []spatialmath.Geometry{box})},
+	}
+
+	// No longer reachable with The Wall in the way
+	err = checkReachable(worldState, constraints)
+	test.That(t, err, test.ShouldNotBeNil)
+
+	// Reachable again if xarm6 and gripper ignore collisions with The Wall
```

Comment:
Worth adding another checkpoint that tests that the goal is still not reachable if only specifying an allowed collision between the VGripper and TheWall?

Reply:
Sure, I can add

Reply:
I think it's of marginal value, probably not worth the effort here tbh.

Reply:
I think its of marginal value, but if you add it I won't mind.